### PR TITLE
Added a more conversion tests

### PR DESCRIPTION
**Pull request by BryanCutler**
tests for bool and byte types to round out implemented types in the conversion so far

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/Arrow.scala [hunk 1]
```diff
@@ -41,7 +41,7 @@ object Arrow {
       case LongType => new ArrowType.Int(8 * LongType.defaultSize, true)
       case FloatType => new ArrowType.FloatingPoint(FloatingPointPrecision.SINGLE)
       case DoubleType => new ArrowType.FloatingPoint(FloatingPointPrecision.DOUBLE)
-      case ByteType => new ArrowType.Int(8, false)
+      case ByteType => new ArrowType.Int(8, true)
```
Comment (BryanCutler) [inline sql/core/src/main/scala/org/apache/spark/sql/Arrow.scala:44]: @icexelloss, I think this should be signed right?  Or is it better to use Arrow binary type for this?

Comment (icexelloss) [reply sql/core/src/main/scala/org/apache/spark/sql/Arrow.scala:44]: Looks like scala treats bytes as signed int:

scala> scala.Byte.MinValue
res2: Byte = -128

scala> scala.Byte.MaxValue
res3: Byte = 127

So yes I think this is right

File: sql/core/src/test/scala/org/apache/spark/sql/ArrowSuite.scala [hunk 2]
```diff
@@ -77,7 +90,16 @@ class ArrowSuite extends SharedSQLContext {
     collectAndValidate(lowerCaseData, "test-data/arrow/lowercase-strings.json")
   }
 
-  test("time and date conversion") { }
+  ignore("time and date conversion") {
```
Comment (icexelloss) [inline sql/core/src/test/scala/org/apache/spark/sql/ArrowSuite.scala:93]: FYI, I am adding support for time and date. I will rebase on this and finish this test.

**Comment by icexelloss** [comment]
LGTM

File: sql/core/src/test/scala/org/apache/spark/sql/ArrowSuite.scala (hunk 2 above)
Comment (BryanCutler) [reply sql/core/src/test/scala/org/apache/spark/sql/ArrowSuite.scala:93]: Great!

**Pull request closed without merging**